influxdb_store.go

```diff
@@ -145,6 +173,85 @@ func (in *InfluxDBStore) Trace(id ID) (*Trace, error) {
 	return trace, nil
 }
 
+func mustJSONFloat64(x interface{}) float64 {
+	n := x.(json.Number)
+	v, err := n.Float64()
+	if err != nil {
+		panic(err)
+	}
+	return v
+}
+
+func mustJSONInt64(x interface{}) int64 {
+	n := x.(json.Number)
+	v, err := n.Int64()
+	if err != nil {
+		panic(err)
+	}
+	return v
+}
+
+// Aggregate implements the Aggregator interface.
+func (in *InfluxDBStore) Aggregate(start, end time.Duration) ([]*AggregatedResult, error) {
+	// Find the mean (average), minimum, maximum, std. deviation, and count of
+	// all spans.
+	q := `SELECT MEAN("duration"),MIN("duration"),MAX("duration"),STDDEV("duration"),COUNT("duration") from spans`
+	q += fmt.Sprintf(
+		" WHERE time >= '%s' AND time <= '%s'",
+		time.Now().Add(start).UTC().Format(time.RFC3339Nano),
+		time.Now().Add(end).UTC().Format(time.RFC3339Nano),
+	)
+	q += ` GROUP BY "name"`
+	result, err := in.executeOneQuery(q)
+	if err != nil {
+		return nil, err
+	}
+
+	// Populate the results.
+	results := make([]*AggregatedResult, len(result.Series))
+	for i, row := range result.Series {
+		v := row.Values[0]
+		mean, min, max, stddev, count := v[1], v[2], v[3], v[4], v[5]
+		results[i] = &AggregatedResult{
+			RootSpanName: row.Tags["name"],
+			Average:      time.Duration(mustJSONFloat64(mean) * float64(time.Second)),
```

Comment:
Could we replace `float64(time.Second)` with a variable instead? declared & assigned around line 209, to avoid repetitive equal calculations.